src/cpu.ts

```diff
@@ -0,0 +1,328 @@
+import { Clock } from "./clock";
+import { Speaker } from "./speaker";
+
+export class CPU {
+  public drawFlag = false;
+  public gfx = new Uint8Array(64 * 32);
+  public key = new Uint8Array(16);
+  private pc = 0x200;
+  private opcode = 0;
+  private I = 0;
+  private sp = 0;
+  private memory = new Uint8Array(4069);
+  private V = new Uint8Array(16);
+  private delayTimer = 0;
+  private soundTimer = 0;
+  private stack: Uint16Array = new Uint16Array(16);
+  private speaker = new Speaker();
+  private timerRef: number | undefined;
+  private clock = new Clock(60);
+
+  constructor(program: Uint8Array) {
+    this.loadFontsetIntoMemory();
+    this.loadProgramIntoMemory(program);
+    this.updateTimer();
+  }
+
+  public cycle() {
+    this.opcode = this.memory[this.pc] << 8 | this.memory[this.pc + 1];
+    this.handleOpcode();
+  }
+
+  public setKeys(keys: Uint8Array) {
+    this.key = keys;
+  }
+
+  public killTimer() {
+    if (this.timerRef) {
+      clearInterval(this.timerRef);
+    }
+  }
+
+  private handleOpcode() {
+    const X = (this.opcode & 0x0F00) >> 8;
```

Comment:
You should probably put those magic numbers into the consts with meaningful names.